aspnetcore/performance/caching/output.md

```diff
@@ -0,0 +1,155 @@
+ms.date: 10/26/2022
+uid: performance/caching/output
+---
+# Output caching middleware in ASP.NET Core
+
+By [Tom Dykstra](https://github.com/tdykstra)
+
+:::moniker range=">= aspnetcore-7.0"
+
+This article explains how to configure output caching middleware in an ASP.NET Core app. For an introduction to output caching, see [Output caching](xref:performance/caching/overview#output-caching).
+
+The output caching middleware supports use in all types of ASP.NET Core apps: Minimal API, Web API with controllers, MVC, and Razor Pages. The sample app is a Minimal API, but every caching feature it illustrates is also supported in the other app types.
+
+## Add the middleware to the app
+
+Add the output caching middleware to the service collection by calling <xref:Microsoft.Extensions.DependencyInjection.OutputCacheServiceCollectionExtensions.AddOutputCache%2A>.
+
+Add the middleware to the request processing pipeline by calling <xref:Microsoft.AspNetCore.Builder.OutputCacheApplicationBuilderExtensions.UseOutputCache%2A>.
+
+> [!NOTE]
+> * In apps that use [CORS middleware](xref:security/cors), `UseOutputCache` must be called after <xref:Microsoft.AspNetCore.Builder.CorsMiddlewareExtensions.UseCors%2A>.
+> * In Razor Pages apps and apps with controllers, `UseOutputCache` must be called after `UseRouting`.
+> * Calling `AddOutputCache`and `UseOutputCache` doesn't start caching behavior. It makes caching available, but the actual caching behavior must be configured as shown in the following sections.
+
+## Configure one endpoint or page
+
+For minimal API apps, configure an endpoint to do caching by calling `CacheOutput()` or by applying the `[OutputCache]` attribute, as shown in the following examples:
+
+:::code language="csharp" source="output/samples/7.x/Program.cs" id="oneendpoint":::
+
+For apps with controllers, apply the `[OutputCache]` attribute to the action method. For Razor Pages apps, apply the attribute to the Razor page class.
+
+## Configure multiple endpoints or pages
+
+Create *policies* when calling `AddOutputCaching` to specify caching configuration that applies to multiple endpoints. A policy can be selected for specific endpoints, while a base policy provides default caching configuration for a collection of endpoints.
+
+The following highlighted code configures caching for all of the app's endpoints, with expiration time of 10 seconds. If you don't explicitly configure an expiration time, it's one minute.
+
+:::code language="csharp" source="output/samples/7.x/Program.cs" id="policies1" highlight="3":::
+
+The following highlighted code creates two policies, each specifying a different expiration time. Selected endpoints can use the 20 second expiration, and others can use the 30 second expiration.
+
+:::code language="csharp" source="output/samples/7.x/Program.cs" id="policies1" highlight="4-5":::
+
+You can select a policy for an endpoint when calling the `CacheOutput` method or using the `[OutputCache]` attribute:
```

Comment:
```suggestion
A policy for an endpoint can be selected when calling the `CacheOutput` method or using the `[OutputCache]` attribute:
```

Reply:
The .NET style guide [advises against use of passive voice](https://learn.microsoft.com/en-us/contribute/dotnet/dotnet-voice-tone#use-active-voice).